src/StoreMeta.h

```diff
@@ -80,67 +64,96 @@ enum {
         uint16_t flags;
      \endcode
      */
-    STORE_META_STD,
-
-    /**
-     * Reserved for future hit-metering (RFC 2227) stuff
-     */
-    STORE_META_HITMETERING,
-
-    // TODO: document this TLV type code
-    STORE_META_VALID,
+    STORE_META_STD = 5,
 
     /**
      * Stores Vary request headers
      */
-    STORE_META_VARY_HEADERS,
+    STORE_META_VARY_HEADERS = 8,
 
     /**
      * Updated version of STORE_META_STD, with support for  >2GB objects.
      * As STORE_META_STD except that the swap_file_sz is a 64-bit integer instead of 32-bit.
      */
-    STORE_META_STD_LFS,
+    STORE_META_STD_LFS = 9,
 
     // TODO: document this TLV type code
-    STORE_META_OBJSIZE,
+    STORE_META_OBJSIZE = 10
 
-    STORE_META_STOREURL,    /**< the Store-ID url, if different to the normal URL */
-    STORE_META_VARY_ID,     /**< Unique ID linking variants */
-    STORE_META_END
+    // When adding values, update Store::SwapMetaTypeMax.
+    // When removing values, check Store::IgnoredSwapMetaType() and friends.
 };
 
-/// \ingroup SwapStoreAPI
-class StoreMeta
+namespace Store {
+
+/// swap meta type ID written to or loaded from Store
+using RawSwapMetaType = char;
+
+/// maximum value of a recognized swap meta type
+const int SwapMetaTypeMax = 12; // Use "inline constexpr ..." with C++17.
+
+/// Store entries with longer swap metadata field values are not swapped out and
+/// are considered invalid when validating being-loaded metadata. This arbitrary
+/// limit protects code that adds individual swap metadata field sizes from
+/// overflowing and also prevents allocation of huge buffers when loading
+/// variable-length fields. Reevaluate this limit when increasing MAX_URL.
+const size_t SwapMetaFieldValueLengthMax = 64*1024;
```

Comment:
Please assert here that MAX_URL will fit. The comment above is not enough to get attention of people casually modifying MAX_URL on ./configure
```suggestion
const size_t SwapMetaFieldValueLengthMax = 64*1024;
static_assert(SwapMetaFieldValueLengthMax >= MAX_URL, "MAX_URL will fit in a Swap meta field");
```